packages/flutter_tools/templates/app_shared/android-java.tmpl/build.gradle.tmpl

```diff
@@ -11,8 +11,8 @@ buildscript {
     }
 }
 
-allprojects {
-    repositories {
+[rootProject, project(':app')].each { project ->
```

Comment:
this looks ugly

Reply:
Probably the better thing would be to add it to `dependencyResolutionManagement` in `settings.gradle`:

```groovy
dependencyResolutionManagement {
    repositories {
        mavenCentral()
        google()
    }
}
```

but I think it'd apply it to all the subprojects (so also other Flutter plugins, which we don't want).

Reply:
I'm pretty sure this will happen then:

> apply the repository to all the subprojects (so also other Flutter plugins, which we don't want).

The main problem here is that Flutter plugins on Android are included as normal Gradle projects. Let's say I'm depending on the [`location`](https://pub.dev/packages/location) plugin in my `pubspec.yaml`. Then the Gradle project hierarchy looks like this:

```
:android
  :app
  :location
```

so `dependencyResolutionManagement` (declared in `:android`'s `settings.gradle`) will also apply to `:location`. This behavior may cause Flutter plugin authors to not include `repositories` they actually use, since the app in the context of which the plugin will work will already have declared the `repositories`. But is it that bad of a problem? I'm not sure.